x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -480,8 +480,12 @@ private void prewarmCache(ActionListener<Void> listener) {
 
                         logger.trace("{} warming cache for [{}] part [{}/{}]", shardId, file.physicalName(), part + 1, numberOfParts);
                         final long startTimeInNanos = statsCurrentTimeNanosSupplier.getAsLong();
-                        ((CachedBlobContainerIndexInput) input).prefetchPart(part);
-                        recoveryState.getIndex().addRecoveredBytesToFile(file.physicalName(), file.partBytes(part));
+                        final long cachedSize = ((CachedBlobContainerIndexInput) input).prefetchPart(part);
```

Comment:
I wonder if we could be more precise here by returning `Tuple<Persistent Cache Length, Prefetched Length>` ?

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/67425/commits/e0c5c8d9eb3b271d669bdedc239ca255b0d510b6